examples/rvc-app/rvc-common/src/rvc-device.cpp

```diff
@@ -104,7 +106,38 @@ void RvcDevice::HandleOpStatePauseCallback(Clusters::OperationalState::GenericOp
 
 void RvcDevice::HandleOpStateResumeCallback(Clusters::OperationalState::GenericOperationalError & err)
 {
-    auto error = mOperationalStateInstance.SetOperationalState(to_underlying(OperationalState::OperationalStateEnum::kRunning));
+    auto error = CHIP_NO_ERROR;
+    switch (mOperationalStateInstance.GetCurrentOperationalState())
+    {
+    case to_underlying(RvcOperationalState::OperationalStateEnum::kCharging):
+    case to_underlying(RvcOperationalState::OperationalStateEnum::kDocked): {
+        if (mRunModeInstance.GetCurrentMode() == RvcRunMode::ModeCleaning ||
+            mRunModeInstance.GetCurrentMode() == RvcRunMode::ModeMapping)
+        {
+            error = mOperationalStateInstance.SetOperationalState(to_underlying(OperationalState::OperationalStateEnum::kRunning));
+        }
+        else
+        {
+            err.Set(to_underlying(OperationalState::ErrorStateEnum::kCommandInvalidInState));
+            return;
+        }
+    }
+    break;
+    case to_underlying(OperationalState::OperationalStateEnum::kPaused): {
+        if (mPausedState == to_underlying(RvcOperationalState::OperationalStateEnum::kSeekingCharger))
+        {
+            error = mOperationalStateInstance.SetOperationalState(
+                to_underlying(RvcOperationalState::OperationalStateEnum::kSeekingCharger));
+        }
+        else
+        {
+            error = mOperationalStateInstance.SetOperationalState(to_underlying(OperationalState::OperationalStateEnum::kRunning));
+        }
+    }
+    break;
+    }
```

Comment:
should there be a default here? what if current running state is neither paused nor docking/charging?
Is that ok to just say "success"? Should have error been initialized with an unsupported error instead?

Reply:
I did consider this but the SDK handles all the states that are not `paused`, `charging` or `docked`. The only other states that are allowed to reach this code are the vendor defined states. However, this example does not define any. However, having said that, clients like chip-tool can still try to set the state to a vendor defined state so it would be useful to fail.

Reply:
Please add a default and log at least some unexpected (and likely set an error because unexpected state). I feel uncomfortable in having a uint8 switch that only handles a subset of cases  and if an unknown state is in we just return "success".